src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -420,43 +420,39 @@ public override int GetHashCode()
 
         public struct Enumerator : IEnumerator<string>
         {
-            private readonly StringValues _values;
+            private readonly string[] _values;
             private string _current;
             private int _index;
 
-            public Enumerator(StringValues values)
+            public Enumerator(ref StringValues values)
             {
-                _values = values;
-                _current = null;
+                _values = values._values;
+                _current = values._value;
```

Comment:
StringValue with 0 and 1 count; where the array is not set. Changed move next to fit the behaviour.